static/app/components/onboardingWizard/newSidebar.tsx

```diff
@@ -1,18 +1,73 @@
-import {Fragment, useState} from 'react';
+import {
+  Fragment,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
 import {css} from '@emotion/react';
 import styled from '@emotion/styled';
+import partition from 'lodash/partition';
 
+import {navigateTo} from 'sentry/actionCreators/navigation';
+import {updateOnboardingTask} from 'sentry/actionCreators/onboardingTasks';
 import {Chevron} from 'sentry/components/chevron';
 import InteractionStateLayer from 'sentry/components/interactionStateLayer';
+import {
+  OnboardingContext,
+  type OnboardingContextProps,
+} from 'sentry/components/onboarding/onboardingContext';
+import {findCompleteTasks, taskIsDone} from 'sentry/components/onboardingWizard/utils';
 import ProgressRing from 'sentry/components/progressRing';
 import SidebarPanel from 'sentry/components/sidebar/sidebarPanel';
 import type {CommonSidebarProps} from 'sentry/components/sidebar/types';
 import {Tooltip} from 'sentry/components/tooltip';
 import {IconCheckmark} from 'sentry/icons';
 import {t, tct} from 'sentry/locale';
+import DemoWalkthroughStore from 'sentry/stores/demoWalkthroughStore';
 import pulsingIndicatorStyles from 'sentry/styles/pulsingIndicator';
 import {space} from 'sentry/styles/space';
+import {type OnboardingTask, OnboardingTaskGroup} from 'sentry/types/onboarding';
+import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+import {trackAnalytics} from 'sentry/utils/analytics';
 import {isDemoWalkthrough} from 'sentry/utils/demoMode';
+import useApi from 'sentry/utils/useApi';
+import useOrganization from 'sentry/utils/useOrganization';
+import useProjects from 'sentry/utils/useProjects';
+import useRouter from 'sentry/utils/useRouter';
+
+import {getMergedTasks} from './taskConfig';
+
+/**
+ * How long (in ms) to delay before beginning to mark tasks complete
+ */
+const INITIAL_MARK_COMPLETE_TIMEOUT = 600;
+
+/**
+ * How long (in ms) to delay between marking each unseen task as complete.
+ */
+const COMPLETION_SEEN_TIMEOUT = 800;
```

Comment:
[This code as copied](https://github.com/getsentry/sentry/blob/2adc755ec95cc44300817e592a1a4db056c00641/static/app/components/onboardingWizard/sidebar.tsx#L33-L41) from the old component.